### PR TITLE
V8 - LogViewer First Draft Doc

### PR DESCRIPTION
**Pull request by warrenbuckley**
First draft for the docs team to clean up - Adds in details about V8 LogViewer

### CODE REVIEW

File: Getting-Started/LogViewer/index-v8.md [hunk 1]
```diff
@@ -0,0 +1,152 @@
+---
+keywords: logging logviewer logs serilog messagetemplates logs v8 version8
+versionFrom: 8.0.0
+---
+
+# Log Viewer
+From version 8 we are now shipping a built in Log Viewer feature, to allow you to filter and view log entries and perform much more complex search queries to help you find the log entries that you are interested in your Umbraco site.
```
Comment (jmayntzhusen) [inline Getting-Started/LogViewer/index-v8.md]:
```suggestion
From version 8 we are now shipping a built in Log Viewer feature, to allow you to filter and view log entries and perform much more complex search queries to help you find the log entries that you are interested in your Umbraco site.
You can find the log viewer in the settings section.
```

File: Getting-Started/LogViewer/index-v8.md [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+
+## Benefits
+Wanted to find all log entries which contains the same reqest ID or find all items in the log where a property called duration is greater than 1000ms. With the power of structured logging and a query language we are able to search and find log items for very specific scenarios and when debugging the client site you should now have more power to see and find patterns in your log files and get rid of those pesky errors.
```
Comment (jmayntzhusen) [inline Getting-Started/LogViewer/index-v8.md]:
```suggestion
Have you ever wanted to find all log entries which contains the same request ID or find all items in the log where a property called duration is greater than 1000ms? 
With the power of structured logging and a query language we are able to search and find log items for very specific scenarios. When debugging the client site you should now have more power to see and find patterns in your log files and get rid of those pesky errors.
```

File: Getting-Started/LogViewer/index-v8.md [hunk 3]
```diff
@@ -0,0 +1,152 @@
...
+
+## Example queries
+Here are a handful example queries to get you started, however the saved searches contain some further examples. For more details on the syntax head over to the https://github.com/serilog/serilog-filters-expressions project.
+
+**Find all logs that are from the namespace 'Umbraco.Core'**<br/>
+`StartsWith(SourceContext, 'Umbraco.Core')`<br/>
+
+**Find all logs that have the property 'Duration' and the duration is greater than 1000ms'**<br/>
```
Comment (jmayntzhusen) [inline Getting-Started/LogViewer/index-v8.md]:
```suggestion
**Find all logs that have the property 'Duration' and the duration is greater than 1000ms**<br/>
```

File: Getting-Started/LogViewer/index-v8.md [hunk 4]
```diff
@@ -0,0 +1,152 @@
...
+`Has(Duration) and Duration > 1000`<br/>
+
+**Find all logs that the message has localhost in it with SQL like**<br/>
```
Comment (jmayntzhusen) [inline Getting-Started/LogViewer/index-v8.md]:
```suggestion
**Find all logs where the message has localhost in it with SQL like**<br/>
```

File: Getting-Started/LogViewer/index-v8.md [hunk 5]
```diff
@@ -0,0 +1,152 @@
...
+`@Message like '%localhost%'`<br/>
+
+## Saved Searches
+When writing a custom query that you wish to use often, it is possible to save this and use the dropdown to re-use your saved search. To add a new saved search, use the search box to type your query and click the star icon. In doing so you can give it a friendly name. The default location of the saved searches are persisted as JSON to the following file: `/config/logviewer.searches.config.js`
+
+## Implementing your own Log Viewer
+With the flexability of Umbraco, we give you the power to implement your own `ILogViewer` where you are able to fetch logs and the saved searched from a different location such as Azure table storage.
```
Comment (jmayntzhusen) [inline Getting-Started/LogViewer/index-v8.md]:
```suggestion
With the flexibility of Umbraco, we give you the power to implement your own `ILogViewer` where you are able to fetch logs and the saved searched from a different location such as Azure table storage.
```

File: Getting-Started/LogViewer/index-v8.md [hunk 6]
```diff
@@ -0,0 +1,152 @@
...
+
+### Create your own implementation
+To easily do this we can implment a base class `LogVieweSourceBase` from `Umbraco.Core.Logging.Viewer` like so.
```
Comment (jmayntzhusen) [inline Getting-Started/LogViewer/index-v8.md]:
```suggestion
To easily do this we can implement a base class `LogViewerSourceBase` from `Umbraco.Core.Logging.Viewer` like so.
```

File: Getting-Started/LogViewer/index-v8.md [hunk 7]
```diff
@@ -0,0 +1,152 @@
...
+* Install Serilog.Sinks.AzureTableStorage from Nuget (Latest pre-release)
+* Update `config/serilog.user.config` with credentials (so logs persist to Azure)
+
+```xml
+<add key="serilog:using:AzureTableStorage" value="Serilog.Sinks.AzureTableStorage" />
+<add key="serilog:write-to:AzureTableStorage.connectionString" value="DefaultEndpointsProtocol=https;AccountName=ACCOUNT_NAME;AccountKey=KEY;EndpointSuffix=core.windows.net" />
+<add key="serilog:write-to:AzureTableStorage.formatter" value="Serilog.Formatting.Compact.CompactJsonFormatter, Serilog.Formatting.Compact" />
+```
+
+For more in depth information about logging and how to configure it, please read the logging documentation
```
Comment (jmayntzhusen) [inline Getting-Started/LogViewer/index-v8.md]:
```suggestion
For more in depth information about logging and how to configure it, please read the [logging documentation](../Code/Debugging/Logging/index-v8).
```

Comment (warrenbuckley) [reply Getting-Started/LogViewer/index-v8.md]: Not sure why I can't use the Commit suggestion feature here - will manually make the change @jmayntzhusen

**Pull request merged**